5-Adversarial-Search/alphaBeta.js

```diff
@@ -1,36 +1,85 @@
-// Code for Alpha Beta Pruning
-function ALPHA_BETA_SEARCH(state) {
-	var largest_value = MAX_VALUE(state, 0, Number.MAX_SAFE_INTEGER);
-	var action_list = actions(state);
-	for (var action = 0; action < action_list.length; i++) {
-		if  (largest_value ==  utillity(RESULT(state, action_list[action]))) {
-			return action_list[action];
+function alphabeta() {
+	let cur = abstates
+	let scale = 600
+	let div = document.getElementById("alphabetaCanvas")
+    let canvas = document.createElementNS("http://www.w3.org/2000/svg", 'svg')
+    canvas.setAttribute('viewBox', '0 0 ' + scale + ' ' + scale*0.30 + ' ')
+    div.appendChild(canvas)
+	let tree = new Tree(new Board([0,0,-1,1,0,0,1,-1,1], -1), 5)
+	equip_graphics(tree, 0, scale, 12, 5, canvas)
+
+	function setup(tree, depth) {
+		let color = 'hsl(0,0%,50%)'
+		for (let i = 0; i < 9; i++) {
+			tree.graphic.cross_marks[i].setAttribute('stroke', color)
+			tree.graphic.circle_marks[i].setAttribute('stroke', color)
+		}
+		tree.graphic.rect.setAttribute('stroke', color)
+		tree.graphic.l1.setAttribute('stroke', color)
+		tree.graphic.l2.setAttribute('stroke', color)
+		tree.graphic.l3.setAttribute('stroke', color)
+		tree.graphic.l4.setAttribute('stroke', color)
+		if (depth == 0)
+			return
+		for (let i = 0; i < tree.children.length; i++) {
+			setup(tree.children[i], depth-1)
+			tree.branches[i].setAttribute('stroke', color)
 		}
 	}
-}
-function MAX_VALUE(state, alpha, beta) {
-	if (terminal(state))
-		return utillity(state);
-	var largest_value = 0;
-	var action_list = actions(state);
-	for (var action = 0; action < action_list.length; action++) {

```

Comment:
JS has this neat iterator, like for (let action of action_lists) which works really nice. worth using it.